server/service/osquery_utils/queries.go

```diff
@@ -1073,6 +1081,52 @@ FROM chrome_extensions`,
 	DirectIngestFunc: directIngestSoftware,
 }
 
+// SoftwareOverrideQueries are used to override software detail query results.  These DetailQueries
+// must include a `SoftwareOverrideMatch` function that returns true if the software row should be
+// overridden with the results of `Query`.
+// Software queries expect specific columns to be present.  Reference the
+// software_{macos|windows|linux} queries for the expected columns.
+var SoftwareOverrideQueries = map[string]DetailQuery{
+	// macos_firefox Differentiates between Firefox and Firefox ESR by checking the RemotingName value in the
+	// application.ini file. If the RemotingName is 'firefox-esr', the name is set to 'Firefox ESR.app'.
+	"macos_firefox": {
+		Query: `
+			WITH app_paths AS (
+				SELECT path 
+				FROM apps 
+				WHERE bundle_identifier = 'org.mozilla.firefox'
+			),	
+			remoting_name AS (
+				SELECT value, path 
+				FROM parse_ini 
+				WHERE key = 'RemotingName' 
+				AND path IN (SELECT CONCAT(path, '/Contents/Resources/application.ini') FROM app_paths)
+			)
+			SELECT
+				CASE
+					WHEN remoting_name.value = 'firefox-esr' THEN 'Firefox ESR.app'
+					ELSE 'Firefox.app'
```

Comment:
What about this query returning no results if Firefox ESR is not installed? Ingestion method will return earlier in:
```sh
	// Extract the results of the extra query.
	softwareExtraRows, _ := (*results)[softwareExtraQuery]
	if len(softwareExtraRows) == 0 {
		return
	}
```

Reply:
the discovery query checks the existence of either version of firefox (both use the same bundleIdentifier), so if the ESR version isn't installed, it will still return "Firefox.app".  There shouldn't be a scenario where this query runs and doesn't return anything.

Reply:
Right, but why do we want to return (non-ESR) `Firefox.app`? (it's already included in the common software list)

Reply:
Let me think about that one, it does seem to make sense that we should only override if ESR is detected.

Reply:
There is an edge case where a host has both ESR and non-ESR versions installed, but  we could delete only existing rows that match the same application path.